Update HA Addon to use image from libretiny-eu

catalin2402 opened the pull request:
To update the Home Assistant addon images, checkout this branch, update it with the mainstream and run these commands

`docker buildx create --use desktop-linux`

`docker buildx build --build-arg BASEIMGTYPE=hassio --build-arg BUILD_VERSION=latest --file docker/Dockerfile --platform linux/amd64,linux/aarch64,linux/arm/v7 --target hassio --tag ghcr.io/libretiny-eu/libretiny-esphome-hassio:latest . --push`

`docker buildx build --build-arg BASEIMGTYPE=docker --build-arg BUILD_VERSION=latest --file docker/Dockerfile --platform linux/amd64,linux/aarch64,linux/arm/v7 --target docker --tag ghcr.io/libretiny-eu/libretiny-esphome-docker:latest . --push`

Commit:
Updated HA Addon to use image from libretiny-eu

Commit:
Fixed docker git version

kuba2k2 commented:
Why are the `libc6` hacks still needed? I thought upgrading to GCC 10.3.1 (which has all native binaries available on PIO registry) would fix the issue (didn't it?).

Doesn't the problem only occur now for people building an armv7 container on an aarch64 machine? (or the equivalent for x86)

catalin2402 commented:
Kept them from the old files of the addon :), I will try to remove them and see if everything is compiling fine on x64, arm64 and armv7

kuba2k2 commented:
Yeah, because if someone is compiling 32bit on a 64bit host, it's usually not right and we probably don't need to support this, which should make the images smaller too.

catalin2402 commented:
I think the difference is something like 10MB, I will test it tomorrow

Commit:
Updated Dockerfile

catalin2402 commented:
Hey, I removed libc6 dependencies, I added the changes from https://github.com/esphome/esphome/commit/57e909e7909b5a11024fe630c506ba166243a68c and now the image is only 300 MB as normal :)

kuba2k2 commented on docker/Dockerfile:
```diff
@@ -152,4 +152,4 @@ COPY requirements_test.txt /
 
 VOLUME ["/esphome"]
-WORKDIR /esphome
+WORKDIR /esphome
```
```suggestion
WORKDIR /esphome

```

kuba2k2 commented on script/setup:
```diff
@@ -15,4 +15,4 @@ pip3 install --no-use-pep517 -e .
 pre-commit install
 
-script/platformio_install_deps.py platformio.ini
+script/platformio_install_deps.py platformio.ini --libraries --tools --platforms
```
```suggestion
script/platformio_install_deps.py platformio.ini --libraries --tools --platforms

```

kuba2k2 commented on script/platformio_install_deps.py:
```diff
@@ -45,4 +55,4 @@
             tools.append(tool)
 
-subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])
+subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])
```
```suggestion
subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])

```

kuba2k2 committed:
Apply suggestions from code review
